plot_explorer.py

```diff
@@ -793,15 +796,35 @@ def closeEvent(self, event):
 
 if __name__ == '__main__':
 
+    path_icon = str(Path(__file__).parent / 'assets/openmc_logo.png')
+    path_splash = str(Path(__file__).parent / 'assets/splash.png')
+
+
     app = QApplication(sys.argv)
     app.setOrganizationName("OpenMC")
     app.setOrganizationDomain("openmc.org")
     app.setApplicationName("OpenMC Plot Explorer")
-    path_logo = Path(__file__).parent / 'openmc_logo.png'
-    app.setWindowIcon(QtGui.QIcon(str(path_logo)))
+    app.setWindowIcon(QtGui.QIcon(path_icon))
     app.setAttribute(QtCore.Qt.AA_DontShowIconsInMenus, True)
 
+    splash_pix = QtGui.QPixmap(path_splash)
+    splash = QSplashScreen(splash_pix, QtCore.Qt.WindowStaysOnTopHint)
+    splash.setPixmap(splash_pix)
+    splash.setMask(splash_pix.mask())
+    splash.showMessage("Starting GUI...", QtCore.Qt.AlignHCenter | QtCore.Qt.AlignBottom)
+    splash.show()
+
+    # this loop sucks, but it makes the splashscreen work
+    start = time.time()
+    timer = 0.0
+    while timer < 0.1:
+        timer = time.time() - start
+        time.sleep(0.001)
+        app.processEvents()
```

Comment:
So what's the deal with this block?

Reply:
Not totally sure tbh, but the splash screen won't display properly without it. My sense is that a delay is needed to either let the `QPixmap` get loaded or to let the `QSplashScreen` initialize.

I've replaced the loop with a delay so at least it isn't hammering that call like it was, but I can dig into this more.

Reply:
With this change, I actually don't see the splash screen any more :confused:

Reply:
Hrmmmm ok I'll dig into this and see if I can come up with something better than a loop/delay.

Reply:
Ok, what I've just pushed is quite a bit more satisfying. Loads the OpenMC model using a `QThread`-derived class and calls `QApplication.processEvents()` until complete. The following 
site recommends calling `proccessEvents` periodically during loading events like this, and the documentation hints at the same. This is apparently because the internal event handling loop doesn't start until `QApplication.exec_()` is called.

http://www.informit.com/articles/article.aspx?p=30708&seqNum=3

Reply:
You can just use a normal Python thread here, which allows you to get rid of the extra class:
```Python
from threading import Thread
...
loader_thread = Thread(target=openmc.capi.init, args=([-c],))
loader_thread.start()
while loader_thread.is_alive():
    app.processEvents()
```

Reply:
Even better. Good call. Updating...